client/my-sites/media-library/upload-button.jsx

```diff
@@ -62,20 +69,19 @@ module.exports = React.createClass( {
 	},
 
 	render: function() {
-		var classes = classNames( 'media-library__upload-button', this.props.className );
+		var classes = classNames( 'media-library__upload-button', 'button', this.props.className );
 
 		return (
-			<Button icon className={ classes }>
-				<form ref="form">
-					{ this.props.children }
-					<input
-						type="file"
-						accept={ this.getInputAccept() }
-						multiple
-						onChange={ this.uploadFiles }
-						className="media-library__upload-button-input" />
-				</form>
-			</Button>
+			<form ref="form" className={ classes }>
```

Comment:
Interesting, I take it that the form wasn't fully filling the button area. If you happen to browser zoom in on IE, then click the bottom right you'll see that nothing happens, but if you click on top left it triggers normally.

If we run into this again later, we can just make the font size absurdly huge on the hidden input.

See http://jsfiddle.net/gwwar/nFLKU/